SocietalConstructionTool/Compiler/IErrorReporter.cs

```diff
@@ -0,0 +1,7 @@
+namespace Sct.Compiler
+{
+    public interface IErrorReporter
+    {
+        IEnumerable<CompilerError> Errors { get; }
```

Comment:
Maybe abstract away to adding errors and getting errors?

Reply:
?